### PR TITLE
Share package.json devDependencies for prod React page builds within the platform repository

### PR DESCRIPTION
**Pull request by cnathe**

### CODE REVIEW

File: assay/build.gradle [hunk 1]
```diff
@@ -3,3 +3,5 @@ import org.labkey.gradle.util.BuildUtils
 dependencies{
     BuildUtils.addLabKeyDependency(project: project, config: "modules", depProjectPath: BuildUtils.getPlatformModuleProjectPath(project.gradle, "experiment"), depProjectConfig: "published", depExtension: "module")
 }
+
+project.tasks.npmInstall.dependsOn(project.project(":server:modules:platform").tasks.npmInstall)
```
Comment (labkey-susanh) [inline assay/build.gradle]: Let's change this to:
```suggestion
project.tasks.npmInstall.dependsOn(project.project(BuildUtils.getPlatformProjectPath(project.gradle).tasks.npmInstall)
```
I don't expect that path to change, but it's nice to have our references centralized.

File: experiment/build.gradle [hunk 2]
```diff
@@ -16,3 +16,4 @@ dependencies{
    BuildUtils.addLabKeyDependency(project: project, config: "modules", depProjectPath: BuildUtils.getPlatformModuleProjectPath(project.gradle, "query"), depProjectConfig: "published", depExtension: "module")
 }
 
+project.tasks.npmInstall.dependsOn(project.project(":server:modules:platform").tasks.npmInstall)
```
Comment (labkey-susanh) [inline experiment/build.gradle]: Same suggestion as above for assay.

**Review by labkey-susanh — CHANGES_REQUESTED** [review]
I like it!

File: assay/build.gradle (hunk 1 above)
Comment (cnathe) [reply assay/build.gradle]: Done. Thanks.

File: experiment/build.gradle (hunk 2 above)
Comment (cnathe) [reply experiment/build.gradle]: Done. Thanks.

**Review by labkey-susanh — APPROVED** [review]

**Pull request merged**